src/server/game/AuctionHouse/AuctionHouseSearcher.cpp

```diff
@@ -0,0 +1,722 @@
+
+        std::shared_ptr<SearchableAuctionEntry> const& auctionEntry = pair.second;
+        auctionEntry->BuildAuctionInfo(searchResponse->packet);
+        ++count;
+        ++totalcount;
+    }
+
+    searchResponse->packet.put<uint32>(0, count);
+    searchResponse->packet << (uint32)totalcount;
+    searchResponse->packet << uint32(AUCTION_SEARCH_DELAY);
+
+    _responseQueue->Enqueue(searchResponse);
+}
+
+void AuctionHouseWorkerThread::SearchBidderListRequest(AuctionSearchBidderListRequest const& searchBidderListRequest)
+{
+    SearchableAuctionEntriesMap const& searchableAuctionMap = GetSearchableAuctionMap(searchBidderListRequest.listFaction);
+
+    AuctionSearcherResponse* searchResponse = new AuctionSearcherResponse();
+    searchResponse->playerGuid = searchBidderListRequest.ownerGuid;
+    searchResponse->packet.Initialize(SMSG_AUCTION_BIDDER_LIST_RESULT, (4 + 4 + 4) + 30000); // pussywizard: ensure there is enough memory
+    searchResponse->packet << (uint32)0;                                     //add 0 as count
+
+    uint32 count = 0;
+    uint32 totalcount = 0;
+
+    for (uint32 const auctionId : searchBidderListRequest.outbiddedAuctionIds)
+    {
+        SearchableAuctionEntriesMap::const_iterator itr = searchableAuctionMap.find(auctionId);
+        if (itr == searchableAuctionMap.end())
+            continue;
+
+        std::shared_ptr<SearchableAuctionEntry> const& auctionEntry = itr->second;
+        auctionEntry->BuildAuctionInfo(searchResponse->packet);
+        ++count;
+        ++totalcount;
+    }
+
+    for (auto const& pair : searchableAuctionMap)
+    {
+        if (pair.second->bidderGuid != searchBidderListRequest.ownerGuid)
+            continue;
+
+        std::shared_ptr<SearchableAuctionEntry> const& auctionEntry = pair.second;
+        auctionEntry->BuildAuctionInfo(searchResponse->packet);
+        ++count;
+        ++totalcount;
+    }
+
+    searchResponse->packet.put<uint32>(0, count);                           // add count to placeholder
+    searchResponse->packet << totalcount;
+    searchResponse->packet << uint32(AUCTION_SEARCH_DELAY);
+
+    _responseQueue->Enqueue(searchResponse);
+}
+
+void AuctionHouseWorkerThread::BuildListAuctionItems(AuctionSearchListRequest const& searchRequest, SortableAuctionEntriesList& auctionEntries, SearchableAuctionEntriesMap const& auctionMap) const
+{
+    // pussywizard: optimization, this is a simplified case
+    if (searchRequest.searchInfo.itemClass == 0xffffffff && searchRequest.searchInfo.itemSubClass == 0xffffffff
+        && searchRequest.searchInfo.inventoryType == 0xffffffff && searchRequest.searchInfo.quality == 0xffffffff
+        && searchRequest.searchInfo.levelmin == 0x00 && searchRequest.searchInfo.levelmax == 0x00
+        && searchRequest.searchInfo.usable == 0x00 && searchRequest.searchInfo.wsearchedname.empty())
+    {
+        for (auto const& pair : auctionMap)
+            auctionEntries.push_back(pair.second.get());
+    }
```

Comment:
Small nitpick: if we add `return` here, we can get rid of the `else` statement below.

Reply:
Done. https://github.com/azerothcore/azerothcore-wotlk/pull/20830/commits/f2e96081fbd46657f48a54a24d81512146600339